pkg/forward/ringbuffer.go

```diff
@@ -0,0 +1,87 @@
+package forward
+
+import (
+	"sync"
+)
+
+// RingBuffer is a fixed-length ring buffer. It can be used by a forwarder, to 'drop' messages instead of applying backpressure. See Issue #15
+type RingBuffer struct {
+	max int
+
+	buf   []string
+	first int          // the index of the first item in the buffer
+	last  int          // the index of the last item in the buffer
+	ch    chan string  // the channel is used to block on the Get method whenever data is unavailable
+	mutex sync.RWMutex // synchronizes changes to buf, first, last
```

Comment:
IMO the presence of both a chan and a mutex indicates a problem in the design: the chan is itself synchronized with a mutex, so it's duplication of effort, in some sense.